### PR TITLE
fix: adjust compose ui

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes mf-2046

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/components/CompositionDialog/Composition.tsx [hunk 1]
```diff
@@ -51,16 +51,31 @@ export function Composition({ type = 'timeline', requireClipboardPermission }: P
     // #region Open
     const [open, setOpen] = useState(false)
     const [isOpenFromApplicationBoard, setIsOpenFromApplicationBoard] = useState(false)
+    const [recoverOverflow, setRecoverOverflow] = useState(false)
+
     const onClose = useCallback(() => {
         setOpen(false)
+        if (recoverOverflow) {
```
Comment (guanbinrui) [inline packages/mask/src/components/CompositionDialog/Composition.tsx]: All dialogs have an overflow problem. Let's patch this in the `<InjectedDialog />` component.

File: packages/shared/src/contexts/components/InjectedDialog.tsx [hunk 2]
```diff
@@ -138,14 +138,36 @@ export function InjectedDialog(props: InjectedDialogProps) {
...
+        }
         onClose?.()
-    }, [isOpenFromApplicationBoard, onClose])
+    }, [isOpenFromApplicationBoard, onClose, recoverOverflow])
+
+    useEffect(() => {
+        const html = document.documentElement
+        if (open && html.style.overflow !== 'hidden') {
+            html.style.overflow = 'hidden'
+            html.style.marginRight = '17px'
```
Comment (Jack-Works) [inline packages/shared/src/contexts/components/InjectedDialog.tsx]: This does not look right. Do you mean to use `scrollLock` from mui `<Modal />` component?

https://mui.com/material-ui/api/modal/

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/shared/src/contexts/components/InjectedDialog.tsx (hunk 2 above)
Comment (lelenei) [reply packages/shared/src/contexts/components/InjectedDialog.tsx]: modal cannot replace dialog

Comment (Jack-Works) [reply packages/shared/src/contexts/components/InjectedDialog.tsx]: what do you mean? dialog is based on modal.

File: packages/shared/src/contexts/components/useScrollLock.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        const html = document.documentElement
+        if (lock && html.style.overflow !== 'hidden') {
+            html.style.overflow = 'hidden'
+            html.style.marginRight = '17px'
+            setRecoverOverflow(true)
+            return
+        }
+        if (!lock && recoverOverflow) {
+            clearScrollLock(true)
+        }
```
Comment (guanbinrui) [inline packages/shared/src/contexts/components/useScrollLock.ts]:
```suggestion
        const body = document.body
        const html = document.documentElement
                
        if (lock) {
            body.style.removeProperty('overflow')
        }

        if (lock && html.style.overflow !== 'hidden') {
            html.style.overflow = 'hidden'
            html.style.marginRight = '17px'
            setRecoverOverflow(true)
            return
        }
        if (!lock && recoverOverflow) {
            clearScrollLock(true)
        }
```

**Pull request merged**